docs/migration_guide.rst

```diff
@@ -1,10 +1,81 @@
+X.X.X migration guide
+=======================
+
+TODO(Update this with correct version or move to separate file)
+
+This version deprecates the AccountClient, which is a major change to the StarkNet.py
+It is being replaced with new :class:`BaseAccount <starknet_py.net.account.base_account.BaseAccount>` ABC and its
+default implementation :class:`Account <starknet_py.net.account.account.Account>`.
+
+Unlike ``AccountClient``, an ``Account`` is not a Client anymore. That means that methods like
+``get_storage_at``, ``call_contract`` etc. are not available in the Account interface.
+
+However, ``Account`` now exposes a `.`client`` property, which means using an ``Account`` is
+just as simple as ``AccountClient`` was. For example:
+
+.. code-block:: python
+
+    # Inspecting storage
+    await account_client.get_storage_at(contract_address=0x1234)
+    # becomes
+    await account.client.get_storage_at(contract_address=0x1234)
+
+
+    call = Call(to_addr=0x1, selector=0x1234, calldata=[])
+
+    # Sending transactions
+    tx = await account_client.sign_invoke_transaction(call, 10000)
+    await account_client.send_transaction(tx)
+    # becomes
+    tx = await account.sign_invoke_transaction(call, max_fee=1000)  # Not that max_fee is now keyword-only argument
+    await account.client.send_transaction(tx)
+
+    # Using execute method
+    await accout_client.execute(call, 1000)
+    # becomes
+    await account.execute(call, max_fee=1000)
+
+
+    # Estimating fee for already created transactions
+    declare_tx = Declare(...)
+    estimate = await account_client.estimate_fee(declare_tx)  # Transaction is implicitly signed
+    # Becomes
+    signature = account.signer.sign_transaction(declare_tx)
+    declare_tx = dataclasses.replace(signature=signature)
```

Comment:
That example won't work :/ I'm wondering if we should create a test case for this entire code and import it as always
@cptartur @Solpatium

Reply:
Hmm, we can do that, though this one example will get removed anyways.